runtime/near-vm-runner/src/wasmtime_runner.rs

```diff
@@ -442,16 +450,30 @@ impl crate::PreparedContract for VMResult<PreparedContract> {
             }
         };
 
-        let memory_copy = memory.0;
+        let engine = module.engine();
+
         let config = Arc::clone(&result_state.config);
-        let mut logic = VMLogic::new(ext, context, fees_config, result_state, &mut memory);
-        let engine = store.engine();
+
+        let ctx = Ctx::default();
+
+        let mut store = Store::<Ctx>::new(engine, ctx);
+        let mut memory = WasmtimeMemory::new(
+            &mut store,
+            config.limit_config.initial_memory_pages,
+            config.limit_config.max_memory_pages,
+        )
+        .expect("failed to construct wasmtime memory");
+        let logic = VMLogic::new(ext, context, fees_config, result_state, &mut memory);
+        #[allow(clippy::missing_transmute_annotations)]
+        store.data_mut().logic.replace(unsafe { transmute(logic) });
```

Comment:
Probably shouldn't be too hard to annotate the types for transmute here.

```
transmute::<VMLogic<'_>, VMLogic<'static>>()
```

also does a great job of explaining that this is only transmuting lifetime away.

---

A brief

```
// SAFE: although the 'static lifetime here is a lie, we are pretty confident that the `VMLogic` here only lives for the duration of the contract method call (which is covered by the original lifetime.)
```

would also go a long way.